packages/mobile/src/app/saga.ts

```diff
@@ -144,39 +156,75 @@ export function* handleDeepLink(action: OpenDeepLink) {
   }
 }
 
-export function* navigatePinProtected(action: NavigatePinProtected) {
+export function* handleNavigatePinProtected(action: NavigatePinProtected) {
   const fornoMode = yield select(fornoSelector)
   try {
+    // TODO: Implement PIN protection for forno mode
     if (!fornoMode) {
-      const pincode = yield call(getPincode, false)
-      yield put(startPinVerification())
-      const account = yield call(getAccount)
-      yield call(web3.eth.personal.unlockAccount, account, pincode, UNLOCK_DURATION)
-      navigate(action.routeName, action.params)
-      yield put(finishPinVerification())
-    } else {
-      // TODO: Implement PIN protection for forno mode
-      navigate(action.routeName, action.params)
+      yield call(getPincode, false)
     }
+    navigate(action.routeName, action.params)
   } catch (error) {
     Logger.error(TAG + '@showBackupAndRecovery', 'Incorrect pincode', error)
     yield put(showError(ErrorMessages.INCORRECT_PIN))
-    yield put(finishPinVerification())
   }
 }
 
 export function* watchNavigatePinProtected() {
-  yield takeLatest(Actions.NAVIGATE_PIN_PROTECTED, navigatePinProtected)
+  yield takeLatest(Actions.NAVIGATE_PIN_PROTECTED, handleNavigatePinProtected)
 }
 
 export function* watchDeepLinks() {
   yield takeLatest(Actions.OPEN_DEEP_LINK, handleDeepLink)
 }
 
+function createAppStateChannel() {
+  return eventChannel((emit: any) => {
+    AppState.addEventListener('change', emit)
+
+    const removeEventListener = () => {
+      AppState.removeEventListener('change', emit)
+    }
+    return removeEventListener
+  })
+}
+
+function* watchAppState() {
+  Logger.debug(`${TAG}@monitorAppState`, 'Starting monitor app state saga')
+  const appStateChannel = yield createAppStateChannel()
+  while (true) {
+    try {
+      const newState = yield take(appStateChannel)
+      Logger.debug(`${TAG}@monitorAppState`, `App changed state: ${newState}`)
+      yield put(setAppState(newState))
+    } catch (error) {
+      Logger.error(`${TAG}@monitorAppState`, error)
```

Comment:
the error should be the 3rd param when calling Logger.error (see it's interface)